homeassistant/components/shelly/__init__.py

```diff
@@ -182,7 +182,7 @@ def _async_device_online(_: Any) -> None:
             data["model"] = device.settings["device"]["type"]
             hass.config_entries.async_update_entry(entry, data=data)
 
-        hass.async_create_task(async_block_device_setup(hass, entry, device))
+        async_block_device_setup(hass, entry, device)
```

Comment:
This is the most significant change in this PR, this method is a callback starts when device is online:
https://github.com/home-assistant-libs/aioshelly/blob/a11631e272a53aee0088e5f739b118809c935598/aioshelly/block_device.py#L139